src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const year4 = 'YYYY';
+  const year2 = 'YY';
+
+  const separatedDate = date.split(fromFormat[3]);
+
+  const dateData = {};
+
+  for (let i = 0; i < separatedDate.length; i++) {
+    dateData[fromFormat[i]] = separatedDate[i];
+  }
+
+  if (!dateData[year2]) {
+    dateData[year2] = dateData[year4].slice(-2);
+  } else {
+    if (dateData[year2] < 30) {
+      dateData[year4] = `20${dateData[year2]}`;
+    } else {
+      dateData[year4] = `19${dateData[year2]}`;
+    }
+  }
+
+  const result = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    result.push(dateData[toFormat[i]]);
+  }
```

Comment:
The loop condition `i < toFormat.length - 1` might cause issues if the `toFormat` array only contains the separator as the last element. You should iterate over all the date parts in `toFormat` and only skip the last element if it is the separator. A safer condition would be `i < toFormat.length && toFormat[i] !== toFormat[3]` to ensure you're only pushing date parts to the result.